### ISSUE TITLE
Add Topics over Time (ToT) for continuous-time topic prevalence

### ISSUE THREAD

**Issue by nealcaren** [nealcaren/topica#694]
## What you want and why

Add **Topics over Time** (ToT; Wang & McCallum, 2006), a topic model that jointly models words and each document's continuous timestamp. It gives each topic a continuous-time density, making it useful for descriptive questions such as when a theme rises, peaks, and fades in an irregularly dated corpus.

This fills a precise temporal gap:

- `DTM` / `DETM` model changes in topic-word content across discrete time slices.
- STM with `spline()` models prevalence conditional on time and remains the recommended route for covariate effects and uncertainty.
- ToT models a topic's continuous temporal distribution directly; it is descriptive rather than causal.

Reference: Xuerui Wang and Andrew McCallum (2006), *Topics over Time: A Non-Markov Continuous-Time Model of Topical Trends*, KDD. https://doi.org/10.1145/1150402.1150450

## Proposed surface

```python
model = topica.TopicsOverTime(num_topics=20, seed=13)
model.fit(docs, timestamps=dates, iters=1000)

model.topic_time        # topics × continuous-time parameters/densities
topica.viz.topic_time(model)
```

Accept Python datetimes, numeric dates, or a documented normalized numeric time vector. Expose a stable per-topic temporal density/summary alongside the existing `topic_word` and `doc_topic` surfaces.

## Acceptance criteria

- Canonical continuous-time ToT inference, with seed-reproducible fitting.
- Input validation and documented normalization for timestamps.
- `topic_word`, `doc_topic`, top-word support, persistence, and representative-document utilities behave consistently with other models.
- A visualization/summary shows per-topic temporal densities and peak times.
- Tests include planted recovery or an independent numerical reference, deterministic-seed behavior, and timestamp-validation cases.
- Documentation explains when to prefer ToT versus DTM/DETM and STM + splines.

## Alternatives

Time-binning plus DTM/DETM is appropriate when vocabulary drift is the target. STM with a time spline is appropriate when estimating covariate effects with intervals. Neither makes irregular continuous-time topical prevalence a first-class fitted object.
